src/GitVersion.Core.Tests/IntegrationTests/ContinuousDeploymentTestScenarios.cs

```diff
@@ -0,0 +1,405 @@
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-just-a-test.1", configuration);
+
+        fixture.Checkout("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.0", configuration);
+
+        fixture.MergeNoFF("feature/just-a-test");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2", configuration);
+
+        fixture.Repository.Branches.Remove("feature/just-a-test");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2", configuration);
+
+        fixture.Repository.DumpGraph();
+    }
+
+    [Test]
+    public void ShouldFallbackToTheVersionOnDevelopLikeTheReleaseWasNeverCreatedWhenReleaseHasBeenCanceled()
+    {
+        // *8f062c7 49 minutes ago(HEAD -> develop)
+        // |\  
+        // | *bda6ba8 52 minutes ago
+        // | *6f5cf19 54 minutes ago
+        // * | 3b20f15 50 minutes ago
+        // |/
+        // *f5640b3 56 minutes ago
+        // *2099a07 58 minutes ago(main)
+
+        var configuration = TestConfigurationBuilder.New
+            .WithVersioningMode(VersioningMode.ContinuousDeployment)
+            .WithoutAnyTrackMergeTargets().Build();
+
+        using var fixture = new EmptyRepositoryFixture();
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.0.1-ci.1", configuration);
+
+        fixture.BranchTo("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.2", configuration);
+
+        fixture.BranchTo("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.2", configuration);
+
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.1", configuration);
+
+        fixture.MergeNoFF("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.4", configuration);
+
+        // cancel the release 1.0.0
+        fixture.Repository.Branches.Remove("release/1.0.0");
+
+        // ❌ expected: "0.1.0-alpha.6"
+        fixture.AssertFullSemver("1.1.0-alpha.4", configuration);
+
+        fixture.Repository.DumpGraph();
+    }
+
+    [Test]
+    public void ShouldConsiderTheMergeCommitFromMainToDevelopWhenReleaseHasBeenShippedToProduction()
```

Comment:
Yep that's the opposite scenario of cancelling or skipping a release. Per definition a release is shipped when it was merged to master and/or tagged with a version number. The last point is dependent how your VersioningMode is (ContinuesDelivery/ContinuesDeployment) IMO. Because I know some scenarios where you are shipping your DLLs from main branch before even the version has been tagged. In other scenarios that is not true. At the end it needs to be configurable.

Reply:
Okay, can you please just add a comment which explains that this is what "shipped to production" means?

Reply:
Done